frontend/test/metabase/scenarios/filters/sql-field-filter-date.cy.spec.js

```diff
@@ -232,35 +119,31 @@ function dateFilterSelector({
   filterValue,
   isFilterRequired = false,
 } = {}) {
+  openDateFilterPicker(isFilterRequired);
+
   switch (filterType) {
     case "Month and Year":
-      openDateFilterPicker(isFilterRequired);
-      setMonthAndYearFilter(filterValue);
+      DateFilter.setMonthAndYear(filterValue);
```

Comment:
Changing all these cases from 2-liners to 1-liners may seem like no big deal.

But it is a big deal. 
Conceptually, this function moves closer to doing only one thing: routing the logic.